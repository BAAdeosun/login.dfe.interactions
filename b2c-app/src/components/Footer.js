import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
