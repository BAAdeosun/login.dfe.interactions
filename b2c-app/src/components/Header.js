import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;