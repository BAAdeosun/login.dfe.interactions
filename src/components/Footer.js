import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer