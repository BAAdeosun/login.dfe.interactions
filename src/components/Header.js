import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header