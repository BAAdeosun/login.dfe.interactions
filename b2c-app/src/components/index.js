import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}