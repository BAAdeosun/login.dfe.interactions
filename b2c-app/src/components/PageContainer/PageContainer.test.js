import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.PageContainer />);
});

it('renders correctly without props', () => {
    const tree = renderer
        .create(<components.PageContainer />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with props to setup the page', () => {

    const content =
        <div>
            <p>paragraph 1</p>
            <p>paragraph 2</p>
        </div>

    const formContent = <form></form>

    const belowFormContent =
        <div>
            <p>paragraph 3</p>
        </div>

    const pageConfig = {
        title: "test_title",
        header: "test_header",
        aboveFormContent: content,
        formContent: formContent,
        submitButtonText: 'submit',
        submitHandler: jest.fn(),
        belowFormContent: belowFormContent
    };

    const tree = renderer
        .create(<components.PageContainer pageConfig={pageConfig} />)
        .toJSON();
    expect(tree).toMatchSnapshot();

    //check page title has been set
    expect(global.window.document.title).toBe(`${pageConfig.title} | National Careers Service`)
});