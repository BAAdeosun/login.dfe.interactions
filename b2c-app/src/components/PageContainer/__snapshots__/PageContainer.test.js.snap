// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props to setup the page 1`] = `
<div>
  <div
    className="govuk-width-container"
  >
    <div
      className="govuk-breadcrumbs"
    >
      <ol
        className="govuk-breadcrumbs__list"
      >
        <li
          className="govuk-breadcrumbs__list-item"
        >
          <a
            className="govuk-breadcrumbs__link"
            href="https://nationalcareers.service.gov.uk/"
          >
            Home
          </a>
        </li>
      </ol>
    </div>
    <main
      className="govuk-main-wrapper"
    >
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <h1
            className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
          >
            test_header
          </h1>
          <div>
            <p>
              paragraph 1
            </p>
            <p>
              paragraph 2
            </p>
          </div>
          <form
            id="customForm"
            noValidate={true}
            onSubmit={[MockFunction]}
          >
            <form />
            <button
              className="govuk-button"
              id="preSubmit"
              type="submit"
            >
              submit
            </button>
          </form>
          <div>
            <p>
              paragraph 3
            </p>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`renders correctly without props 1`] = `null`;
